typings/index.d.ts

```diff
@@ -2333,6 +2332,8 @@ export class GuildManager extends CachedManager<Snowflake, Guild, GuildResolvabl
 export class GuildMemberManager extends CachedManager<Snowflake, GuildMember, GuildMemberResolvable> {
   public constructor(guild: Guild, iterable?: Iterable<unknown>);
   public guild: Guild;
+  public add(user: UserResolvable, options: AddGuildMemberOptions & { fetchWhenExisting: true }): Promise<GuildMember>;
+  public add(user: UserResolvable, options: AddGuildMemberOptions): Promise<GuildMember | null>;
```

Comment:
Probably needs to be done this way? `fetchWhenExisting` defaults to `true`, so we need to check for a required `false` value, then the rest of overloads.
```suggestion
  public add(user: UserResolvable, options: AddGuildMemberOptions & { fetchWhenExisting: false }): Promise<GuildMember | null>;
  public add(user: UserResolvable, options: AddGuildMemberOptions): Promise<GuildMember>;
```

You can also add a type test.